contracts/eoseosrateio/eoseosrateio.cpp

```diff
@@ -67,99 +60,188 @@ CONTRACT eoseosrateio : public contract {
+                if(row.transparency){
+                    bp_rate->transparency = (bp_rate->transparency + row.transparency)/2;
+                }
+                row.transparency = bp_rate->transparency;
+                counter++;
+            }
+
+            if (bp_rate->infrastructure){
+                sum += bp_rate->infrastructure;
+                if(row.infrastructure){
+                    bp_rate->infrastructure = (bp_rate->infrastructure + row.infrastructure)/2;
+                }
+                row.infrastructure = bp_rate->infrastructure;
+                counter++;
+            }
+
+            if (bp_rate->trustiness){ 
+                sum += bp_rate->trustiness;
+                if(row.trustiness){
+                    bp_rate->trustiness = (bp_rate->trustiness + row.trustiness)/2;
+                }
+                row.trustiness = bp_rate->trustiness;
+                counter++;
+            }
+
+            if (bp_rate->development){
+                sum += bp_rate->development;
+                if(row.development){
+                    bp_rate->development  = (bp_rate->development + row.development)/2;
+                }
+                row.development = bp_rate->development;
+                counter++;
+            }
+
+            if (bp_rate->community){
+                sum += bp_rate->community;
+                if(row.community){
+                    bp_rate->community = (bp_rate->community + row.community)/2;
+                }
+                row.community = bp_rate->community;
+                counter++;                
+            }
+
+            if(counter){
+                row.proxy_voters_cntr++;
+                row.average =( (sum/counter) + row.average ) /2; 
+                row.updated_at = current_time();
+            }		
          });
       }
     }
 
     // for dev only
-    ACTION erase() {
+    ACTION erase(string table) {
+      // table = bps or stats
       //only contract owner can erase table
       require_auth(_self);
 
-      producers_table bps(_code, _code.value);
-      auto itr = bps.begin();
-      while ( itr != bps.end()) {
-          itr = bps.erase(itr);
+      if(!table.compare("bps")){
+      	producers_table bps(_self, _self.value);
+      	auto itr = bps.begin();
+      	while ( itr != bps.end()) {
+      	    itr = bps.erase(itr);
+      	}
       }
 
-      producers_stats_table bps_stats(_code, _code.value);
-      auto itr_stats = bps_stats.begin();
-      while ( itr_stats != bps_stats.end()) {
-          itr_stats = bps_stats.erase(itr_stats);
+      if(!table.compare("stats")){
+      	producers_stats_table bps_stats(_self, _self.value);
+      	auto itr_stats = bps_stats.begin();
+      	while ( itr_stats != bps_stats.end()) {
+            itr_stats = bps_stats.erase(itr_stats);
+      	}
       }
     }
 
   private:
     TABLE block_producers_stats {
       name bp;
       uint32_t proxy_voters_cntr;
+      float average;
```

Comment:
@kriskoin it is a new value, is necessary to clean previous block_producers_stats's rows?

Reply:
@rubenabix  due to the test ,the table has been cleaned.

Reply:
the contract was testes and now is publish on the junglenet